soc/xtensa/esp32s2/Kconfig.soc

```diff
@@ -19,6 +19,22 @@ config ESPTOOLPY_FLASHFREQ_80M
 	bool
 	default y
 
+config FLASH_SIZE
+	int
+	default $(dt_node_reg_size_int,/soc/flash-controller@3f402000/flash@0,0)
+
+config FLASH_BASE_ADDRESS
+	hex
+	default $(dt_node_reg_addr_hex,/soc/flash-controller@3f402000/flash@0)
+
+config ESPTOOLPY_FLASHSIZE
+	string
+	default "1MB" if FLASH_SIZE = 0x100000
+	default "2MB" if FLASH_SIZE = 0x200000
+	default "4MB" if FLASH_SIZE = 0x400000
+	default "8MB" if FLASH_SIZE = 0x800000
+	default "16MB" if FLASH_SIZE = 0x1000000
```

Comment:
you don't need to do this in Kconfig.

In CMake, this would be a single line of the form:
```
math(EXPR esptoolpy_flashsize "${CONFIG_FLASH_SIZE} / 0x100000")
```

Reply:
Interesting. I'll fix this and will also update this to other SoCs.